airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -107,8 +108,8 @@ def __init__(
                 concurrency_level // 2, 1
             )  # Partition_generation iterates using range based on this value. If this is floored to zero we end up in a dead lock during start up
         else:
-            concurrency_level = self.SINGLE_THREADED_CONCURRENCY_LEVEL
-            initial_number_of_partitions_to_generate = self.SINGLE_THREADED_CONCURRENCY_LEVEL
+            concurrency_level = self._LOWEST_SAFE_CONCURRENCY_LEVEL
```

Comment:
So in this case, can we ever run this in a single threaded fashion? I don't have any examples, but if there was a hypothetical connector that only allows for one active request at a time, how might we be able to support that?

Would in our ideal world there be a separate thread management for generating/enqueuing partitions and consuming partitions? but because we don't have that we just need to have this at a minimum of 2?

Reply:
I think it should be pretty simple: we just need to add a lock in the `PartitionReader` when in a `single thread` mode. I haven't prioritize that because we didn't have a case until we want to support the case you mentioned above or custom cursors.

Reply:
understood. thanks for explaining what that would look like